### PR TITLE
feat: makes inference server details link clickable

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

When the service is running, makes the url link clickable. It will open the default browser 

### Screenshot / video of UI

image

https://github.com/user-attachments/assets/fc160c27-cbaa-4682-8165-1d21f2d38300

### What issues does this PR fix or reference?

Fixes https://github.com/containers/podman-desktop-extension-ai-lab/issues/1213

### How to test this PR?

- [x] unit tests has been provided

**manually**

Go to inference service details, click on the link

### CODE REVIEW

**Review by lstocchi — APPROVED** [review]
LGTM

File: packages/frontend/src/pages/InferenceServerDetails.svelte [hunk 1]
```diff
@@ -192,10 +199,15 @@ export function goToUpPage(): void {
...
                 <div class="flex flex-row gap-4">
-                  <div
-                    class="bg-[var(--pd-label-bg)] text-[var(--pd-label-text)] rounded-md p-2 flex flex-row w-min h-min text-xs text-nowrap items-center">
-                    http://localhost:{service.connection.port}/v1
-                  </div>
+                  {#if service.status === 'running' && service.type === InferenceType.LLAMA_CPP}
+                    <button
+                      on:click={() =>
+                        service && studioClient.openURL(`http://localhost:${service.connection.port}/docs`)}
+                      class="bg-charcoal-600 rounded-md p-2 flex flex-row w-min h-min text-xs text-nowrap items-center underline">
```
Comment (lstocchi) [inline packages/frontend/src/pages/InferenceServerDetails.svelte:206]: Maybe using dynamic bg for light mode?

Comment (axel7083) [inline packages/frontend/src/pages/InferenceServerDetails.svelte:206]:
```suggestion
                      class="bg-[var(--pd-label-bg)] text-[var(--pd-label-text)] rounded-md p-2 flex flex-row w-min h-min text-xs text-nowrap items-center underline">
```

File: packages/frontend/src/pages/InferenceServerDetails.svelte [hunk 2]
```diff
@@ -192,10 +199,15 @@ export function goToUpPage(): void {
               <div class="bg-[var(--pd-content-card-bg)] rounded-md w-full px-4 pt-2 pb-4 mt-2">
                 <span class="text-sm text-[var(--pd-content-card-text)]">Server</span>
                 <div class="flex flex-row gap-4">
-                  <div
-                    class="bg-[var(--pd-label-bg)] text-[var(--pd-label-text)] rounded-md p-2 flex flex-row w-min h-min text-xs text-nowrap items-center">
-                    http://localhost:{service.connection.port}/v1
-                  </div>
+                  {#if service.status === 'running' && service.type === InferenceType.LLAMA_CPP}
```
Comment (jeffmaury) [inline packages/frontend/src/pages/InferenceServerDetails.svelte:202]: I think that the value of the URL should be computed by the Inference server implementation (llama_cpp,...) as the /docs suffix is not generic. And return null will condition the presence of the button without llama_cpp specific code here

**Comment by slemeur** [comment]
I'm worry this change is making the use of the inference server with other tool (application code, other LLMs tools) harder. 
As a developer, I'd like to be able to very quickly copy the base URL of the inference server so I can paste it in my code. If I need documentation on the API end points, then, I'd like to have a direct access to the API documentation. 

As a result, I'd be in favor of having a different button to access the documentation and let the user still be able to "copy" the URL of the inference server.

File: packages/frontend/src/pages/InferenceServerDetails.svelte (hunk 2 above)
Comment (axel7083) [reply packages/frontend/src/pages/InferenceServerDetails.svelte:202]: I don't see an easy way of doing this, as the LlamaCpp Inference provider is a worker, not a manager that can be reused

Comment (jeffmaury) [reply packages/frontend/src/pages/InferenceServerDetails.svelte:202]: Maybe the connection field should be expanded to contain not only the port

**Comment by axel7083** [comment]
Thanks to https://github.com/containers/podman-desktop-extension-ai-lab/pull/1443 inference server have a labels property, which is used to have an optional `api` or `docs` label.

File: packages/frontend/src/pages/InferenceServerDetails.svelte (hunk 2 above)
Comment (axel7083) [reply packages/frontend/src/pages/InferenceServerDetails.svelte]: The value of the field is now the responsability of the Inference Provider!

**Pull request merged**